library/core/src/ops/range.rs

```diff
@@ -1141,6 +1141,12 @@ impl<'a, T: ?Sized + 'a> RangeBounds<T> for (Bound<&'a T>, Bound<&'a T>) {
     }
 }
 
+/// This impl intentionally does not have `T: ?Sized` due to type inference issues.
+// see https://github.com/rust-lang/rust/pull/61584 for discussion of those issues
+///
+/// If you need to use this impl where `T` is unsized,
+/// consider using the `RangeBounds` impl for a 2-tuple of [`Bound<&T>`][Bound],
+/// e.g. replace `start..` with `(Bound::Included(start), Bound::Unbounded)`.
```

Comment:
Few small nits/thoughts:

* I don't think the "due to type inference issues" part needs to be in public docs. That reasoning is a bit of an internal detail, I think it's fine to just make the whole first sentence a non-doc comment.
* Maybe "impl" -> "implementation" for prose
* `e.g.` -> `i.e.` probably works because it's showing a pattern rather than a specific example
* Looks like the paragraph needs a rewrap (100 chars)

Reply:
* Okay, I've made that part a non-doc comment, and made it not mention the reasoning, just link to it.
* :+1:
* :+1:
* The longest of the added lines in the source is ~86 characters; do you mean the length in the rendered docs?

Reply:
Ah sorry, yeah the length is fine but it just looks like it got some random/copy+paste wrapping (e.g. there's plenty of room for "consider using the `RangeBounds`" the next line up). Not going to block on that though, the rest lgtm.

Thanks for the updates!